### DISCUSSION TITLE
how to specify the file association of the file?

### DISCUSSION THREAD

**Discussion by OriLiMu** [sxyazi/yazi#1164] [1. Q&A] [answered]
For example I want to open html file by google-chrome. 
I check the `yazi.toml`

The related code shuold be like:
`{ mime = "text/html",  use = [ "edit", "reveal" ] },`
	
But the `edit, open, reveal` seems cant specify the program by file type. How to fix it?

```toml
edit = [
	{ run = '$EDITOR "$@"', block = true,  for = "unix" },
	{ run = 'code "%*"',    orphan = true, for = "windows" },
]
open = [
	{ run = 'xdg-open "$@"',                desc = "Open", for = "linux" },
	{ run = 'open "$@"',                    desc = "Open", for = "macos" },
	{ run = 'start "" "%1"', orphan = true, desc = "Open", for = "windows" }
]
reveal = [
	{ run = 'open -R "$1"',                          desc = "Reveal", for = "macos" },
	{ run = 'explorer /select, "%1"', orphan = true, desc = "Reveal", for = "windows" },
	{ run = '''exiftool "$1"; echo "Press enter to exit"; read''', block = true, desc = "Show EXIF", for = "unix" },
]
```

**Comment by lpnh** [comment]
Not sure if this applies for your context, but I use `mimeapps.list` in my environment. It's a simple file for configuring default applications for MIME types. Here's more information about it: https://wiki.archlinux.org/title/XDG_MIME_Applications#mimeapps.list

**Reply by OriLiMu** [reply]
> Not sure if this applies for your context, but I use `mimeapps.list` in my environment. It's a simple file for configuring default applications for MIME types. Here's more information about it: https://wiki.archlinux.org/title/XDG_MIME_Applications#mimeapps.list

Thanks. It worked for some situation...

```toml
	{ on = [ "l" ],         run = "open",                                                 desc = "Open the selected files" },
	{ on = [ "<Enter>" ],   run = "open",                                                 desc = "Open the selected files" },
```


here is my setting. Enter works fine, but <kbd>l</kbd> wont work. do you know why ?

**Reply by lpnh** [reply, marked as answer]
Is this from your `keymap.toml`? It looks like an attempt to achieve something similar to the [smart enter](https://yazi-rs.github.io/docs/tips#smart-enter) plugin. Even if you don't want to go with smart enter, you would need to override the default `l` key. You can check how to do that [here](https://yazi-rs.github.io/docs/configuration/overview#mixing).

By the way, probably you don't need this line, because it's already the default one:

```
{ on = [ "<Enter>" ],   run = "open",                                                 desc = "Open the selected files" },
```
